lib/github/oauth.js

```diff
@@ -94,31 +77,30 @@ module.exports = (robot) => {
 
     robot.log({ gitHubUser, slackUser }, 'Linked User');
 
-    req.session.slackUserId = slackUser.id;
-
-    const redirectUrl = `https://slack.com/app_redirect?team=${teamSlackId}&channel=${channelSlackId}`;
-    if (command) {
-      await axios.post(command.response_url, {
-        response_type: 'ephemeral',
-        attachments: [new SignInConfirm(userSlackId, user.html_url, user.login).getAttachment()],
-      });
-
-      if (command.text.startsWith('signin')) {
-        return res.redirect(redirectUrl);
-      } else if (command.text.startsWith('signout')) {
-        return res.redirect(redirectUrl);
-      }
-      return res.redirect(`/slack/command?trigger_id=${command.trigger_id}`);
-    }
     await workspace.client.chat.postEphemeral({
       channel: channelSlackId,
       user: userSlackId,
       attachments: [new SignInConfirm(userSlackId, user.html_url, user.login).getAttachment()],
     });
-    if (state.slackAction && /unfurl-\d+/.test(state.slackAction.callback_id)) {
+
+    if (state.replaySlashCommand) {
+      const command = await PendingCommand.find(state.trigger_id);
+
+      if (!command) {
+        // FIXME: handle issue with expired command
+        throw new Error();
```

Comment:
Not sure if the FIXME will be addressed in this PR. If not, I wonder if we could pass a little bit more information as part of the exception.

Reply:
Good shout! Yeah I ended up just addressing this too in this PR 😄